arbnode/transaction_streamer.go

```diff
@@ -1023,6 +1023,15 @@ func (s *TransactionStreamer) WriteMessageFromSequencer(
 	}
 
 	s.broadcastMessages([]arbostypes.MessageWithMetadataAndBlockHash{msgWithBlockHash}, pos)
+
+	//  If light client reader and espresso client are set, then we need to store the pos in the database
+	//  to be used later to submit the message to hotshot for finalization.
+	if s.lightClientReader != nil && s.espressoClient != nil {
+		//  Only submit the transaction if escape hatch is not enabled
+		if s.shouldSubmitEspressoTransaction() {
+			return s.enqueuePendingTransaction(pos)
+		}
+	}
```

Comment:
Can you confirm these code can be only called by batch poster? I checked all the callers of `WriteMessageFromSequencer` are from sequencer. Confused about why CI didn't complain anything

Reply:
yes, I am going to test this today